src/amt/CIRAHandler.ts

```diff
@@ -135,15 +124,17 @@ export class CIRAHandler {
         if (this.rawChunkedData.includes('401 Unauthorized') && (this.rawChunkedData.includes('</html>'))) {
```

Comment:
I'm not sure about how we are handling this. To me, we should first check the end of the message, then the status code, then process it:
```
if (this.rawChunkedData.includes('0\r\n\r\n')) {
if(statusCode == 401){
...
}else if(statusCode != 200)
..reject
}else{
..resolve
}
```